src/main/scala/li/cil/oc/server/component/traits/ItemInventoryControl.scala

```diff
@@ -7,21 +7,21 @@ import li.cil.oc.api.machine.Context
 import li.cil.oc.util.ExtendedArguments._
 import li.cil.oc.util.InventoryUtils
 import li.cil.oc.util.ResultWrapper.result
-import net.minecraft.inventory.IInventory
 import net.minecraft.item.ItemStack
+import net.minecraftforge.items.IItemHandler
 
 trait ItemInventoryControl extends InventoryAware {
   @Callback(doc = "function(slot:number):number -- The size of an item inventory in the specified slot.")
   def getItemInventorySize(context: Context, args: Arguments): Array[AnyRef] = {
-    withItemInventory(args.checkSlot(inventory, 0), itemInventory => result(itemInventory.getSizeInventory))
+    withItemInventory(args.checkSlot(inventory, 0), itemInventory => result(itemInventory.getSlots))
   }
 
   @Callback(doc = "function(inventorySlot:number, slot:number[, count:number=64]):number -- Drops an item into the specified slot in the item inventory.")
   def dropIntoItemInventory(context: Context, args: Arguments): Array[AnyRef] = {
     withItemInventory(args.checkSlot(inventory, 0), itemInventory => {
       val slot = args.checkSlot(itemInventory, 1)
       val count = args.optItemCount(2)
-      result(InventoryUtils.extractAnyFromInventory(InventoryUtils.insertIntoInventorySlot(_, itemInventory, None, slot), inventory, null, count))
+      result(InventoryUtils.extractAnyFromInventory(InventoryUtils.insertIntoInventorySlot(_, itemInventory, slot), inventory, null, count))
```

Comment:
While we're fixing this hot garbage, this should IMHO also be
```scala
result(InventoryUtils.extractFromInventorySlot(InventoryUtils.insertIntoInventorySlot(_, itemInventory, slot), inventory, null, selectedSlot, count))
```
i.e. only try to insert the *selected* stack in an agent's inventory. Not try each and every item in its inventory until one fits into the item inventory.